### PR TITLE
Configure Greeting Features

### PR DESCRIPTION
**Pull request by lily4178993**
## **Overview:**

This PR implements a range of updates and enhancements aimed at refining the development experience and user interface for the _my_rails_react_webpack_ project.

## **Updates:**

- **Setup and Initial Configurations:**
  - Introduced Babel and Webpack setups to embrace modern JavaScript capabilities and optimize the asset compilation workflow.

- **Codebase Optimization:**
  - Purged unnecessary files to streamline the codebase and enhance maintainability.

- **Data Modeling and Backend Setup:**
  - Created the Message model to manage greeting messages, enriching the app with dynamic content capabilities.
  - Established a root controller and view as the frontend's initial touchpoint.
  - Developed a Greeting controller and API endpoint to facilitate backend-frontend interactions.
  - Seeded the database with a default Greeting message to guarantee immediate functionality.
  - Structured a Messages table to systematically store greeting messages.

- **State Management and Component Interaction:**
  - Adopted Redux for comprehensive state management, enabling efficient data handling throughout the app.
  - Configured the Redux store to centralize state control, aiding in predictable state transitions.
  - Deployed greetingSlice for retrieving random greetings, demonstrating Redux's handling of asynchronous operations.
  - Crafted a Greeting component to exhibit greeting messages, bolstering user interaction.

- **UI Enhancements:**
  - Enhanced the Greeting component by integrating a refresh button, providing a seamless method for users to obtain new greetings.

mobile screenshot

- **Project Documentation and Licensing:**
  - Incorporated the MIT License to affirm the project's open use.
  - Revised the README to detail project setup and information, facilitating an easy start for new contributors.

**Setup Instructions:**
- Refer to the README document for guidance on how to setup the project.

---

Please review the submitted changes and provide your insights.

### CODE REVIEW

File: Gemfile
Comment (franclobo) [inline Gemfile:1]: - [ ] It looks like there is an error processing your Gemfile, please 🙏🏽 make sure that you can run your app using `bin/dev` 👇🏽 

```bash
bin/dev
Error loading RubyGems plugin "/home/franclobo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/plugins/yard_plugin.rb": cannot load such file -- /home/franclobo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/yard-0.9.34/lib/rubygems_plugin.rb (LoadError)
Installing foreman...
Error loading RubyGems plugin "/home/franclobo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/plugins/yard_plugin.rb": cannot load such file -- /home/franclobo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/yard-0.9.34/lib/rubygems_plugin.rb (LoadError)
Fetching foreman-0.87.2.gem
Successfully installed foreman-0.87.2
1 gem installed
14:09:59 web.1  | started with pid 78951
14:09:59 js.1   | started with pid 78952
14:10:00 js.1   | yarn run v1.22.19
14:10:00 js.1   | $ webpack --config webpack.config.js --mode production --watch
14:10:01 web.1  | C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:540:in `materialize': Could not find rails-7.1.3, sprockets-rails-3.4.2, pg-1.5.5, puma-6.4.2, jsbundling-rails-1.3.0, turbo-rails-2.0.3, stimulus-rails-1.3.3, jbuilder-2.11.5, rubocop-1.60.2, tzinfo-data-1.2024.1, bootsnap-1.18.3, debug-1.9.1, web-console-4.2.1, capybara-3.40.0, selenium-webdriver-4.17.0, actioncable-7.1.3, actionmailbox-7.1.3, actionmailer-7.1.3, actionpack-7.1.3, actiontext-7.1.3, actionview-7.1.3, activejob-7.1.3, activemodel-7.1.3, activerecord-7.1.3, activestorage-7.1.3, activesupport-7.1.3, railties-7.1.3, sprockets-4.2.1, nio4r-2.7.0, json-2.7.1, parallel-1.24.0, parser-3.3.0.5, regexp_parser-2.9.0, msgpack-1.7.2, irb-1.11.2, reline-0.4.2, bindex-0.8.1, addressable-2.8.6, mini_mime-1.1.5, rack-3.0.9, xpath-3.2.0, base64-0.2.0, rubyzip-2.3.2, websocket-1.2.10, websocket-driver-0.7.6, zeitwerk-2.6.13, net-imap-0.4.10, net-smtp-0.4.0.1, rails-dom-testing-2.2.0, rack-session-2.0.0, globalid-1.2.1, timeout-0.4.1, bigdecimal-3.1.6, concurrent-ruby-1.2.3, connection_pool-2.4.1, drb-2.2.0, minitest-5.22.2, mutex_m-0.2.0, rackup-2.1.0, rake-13.1.0, thor-1.3.0, rdoc-6.6.2, io-console-0.7.2, public_suffix-5.0.4, net-protocol-0.2.2, loofah-2.22.0, webrick-1.8.1, psych-5.1.2, stringio-3.1.0 in locally installed gems (Bundler::GemNotFound)
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:200:in `specs'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:266:in `specs_for'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/runtime.rb:18:in `setup'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler.rb:162:in `setup'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/setup.rb:23:in `block in <top (required)>'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/ui/shell.rb:159:in `with_level'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/ui/shell.rb:111:in `silence'
14:10:01 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/setup.rb:23:in `<top (required)>'
14:10:01 web.1  |       from <internal:C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
14:10:01 web.1  |       from <internal:C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
14:10:01 web.1  |       from //wsl.localhost/Ubuntu/home/franclobo/microverse/my_rails_react_webpack_app/config/boot.rb:3:in `<top (required)>'
14:10:01 web.1  |       from bin/rails:3:in `require_relative'
14:10:01 web.1  |       from bin/rails:3:in `<main>'
14:10:02 web.1  | exited with code 1
14:10:02 system | sending SIGTERM to all processes
14:10:02 js.1   | exited with code 1
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @lily4178993, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Use webpack ✔️ 
- Use jsbundling-rails ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile
Comment (lily4178993) [reply Gemfile:1]: Hello @franclobo, thank you for your feedback.
Unfortunately, I'm unable to use bin/dev on Windows, which is why I set up the webpack.config file. To run the app locally, the command is "npm run start," as mentioned in the readme file.
Could you assist me in finding a similar command?

File: README.md [hunk 1]
```diff
@@ -1,24 +1,139 @@
...
+rails db:migrate
+rails db:seed
+```
+
+### Usage <a name="usage"></a>
+
+Start the development server:
+
+```bash
+npm run start
```
[the remark this answers is not in the document]
Comment (lily4178993) [reply README.md:105]: Hello @franclobo, thank you for your feedback.
Unfortunately, I'm unable to use bin/dev on Windows, which is why I set up the webpack.config file. To run the app locally, the command is "npm run start," as mentioned in the readme file.
Could you assist me in finding a similar command?

File: Gemfile
Comment (prolajumokeoni) [reply Gemfile:1]: Hi @lily4178993, I ran this command ` rake assets:precompile` and  removed .exe from ` #!/usr/bin/env ruby.exe `and was able to run your app using ` ./bin/dev`
mm

Comment (lily4178993) [reply Gemfile:1]: Ok, Give me few seconds to push the latest changes

**Review by prolajumokeoni — APPROVED** [review]
Hi @lily4178993 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 -  ./bin/dev used to start app ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thank you, @prolajumokeoni, for approving. Could you please clarify the location of `#!/usr/bin/env ruby.exe` in my project structure? I can't seem to locate it.

**Pull request merged**